dbt_platform_helper/providers/terraform_manifest.py

```diff
@@ -159,15 +159,41 @@ def _add_environment_locals(terraform: dict, app: str):
         }
 
     @staticmethod
-    def _add_moved(terraform):
+    def _add_moved(terraform, platform_config):
+        extensions_comment = "Moved extensions-tf to just extensions - this block tells terraform this. Can be removed once all services have moved to the new naming."
         terraform["moved"] = [
             {
-                "//": "Moved extensions-tf to just extensions - this block tells terraform this. Can be removed once all services have moved to the new naming.",
+                "//": extensions_comment,
                 "from": "module.extensions-tf",
                 "to": "module.extensions",
             }
         ]
 
+        extensions = platform_config.get("extensions", {})
+        s3_extension_names = [
+            extension_name
+            for extension_name, extension in extensions.items()
+            if extension["type"] == "s3"
+        ]
+        s3_comment = "S3 bucket resources are now indexed. Can be removed once all services have moved to terraform-platform-modules 5.x."
+
+        for name in s3_extension_names:
+            resources = [
+                "aws_s3_bucket_server_side_encryption_configuration.encryption-config",
+                "aws_s3_bucket_policy.bucket-policy",
+                "aws_kms_key.kms-key",
+                "aws_kms_alias.s3-bucket",
+            ]
+            moves = [f'module.extensions.module.s3["{name}"].{resource}' for resource in resources]
+            for move in moves:
+                terraform["moved"].append(
+                    {
+                        "//": s3_comment,
+                        "from": move,
+                        "to": f"{move}[0]",
+                    }
+                )
```

Comment:
I wonder whether it's worth having two methods now, with names asking it clear what each is adding.

Reply:
I had thought of doing that, but it means checking in each to see if the "moved" block is there yet and appending to it if not. Happy to do that if you think it's worth it, but I suspect we'll be able to remove all these at the same time.

Reply:
OK.